models/Challenge.js

```diff
@@ -26,155 +26,38 @@ class Challenge {
 
-    return new Promise((resolve, reject) => {
-      const client = utils.db.createClient();
-      let returnCode;
-
-      client.connect();
-      client.on('error', (err) => reject(err));
-
-      const query = client.query(sql, [userId, performanceId, spotId]);
-
-      query.on('row', ({ make_challenge }) => {
-        returnCode = parseInt(make_challenge, 10);
-      });
-
-      query.on('end', () => {
-        client.end();
-        resolve(returnCode);
-      });
-
-      query.on('error', (err) => {
-        client.end();
-        reject(err);
-      });
-    });
+    return db.query(sql, [userId, performanceId, spotId], codeFromRow);
   }
 
   static findAllChallengeablePeopleForUser(user, performanceId) {
-    // eslint-disable-next-line
+    if (!performanceId) {
+      return Promise.resolve(null);
+    }
     const sql = queries.challengeablePeople;
 
-    return new Promise((resolve, reject) => {
-      if (!performanceId) {
-        resolve(null);
-      }
-
-      const client = utils.db.createClient();
-      const users = [];
-
-      client.connect();
-      client.on('error', (err) => reject(err));
-
-      const query = client.query(sql, [user.instrument, user.part, user.spotId, user.nameNumber, performanceId]);
-
-      query.on('row', ({ challengedcount, name, spotid, open }) => {
-        users.push(new ChallengeableUser(challengedcount, name, spotid, open));
-      });
-
-      query.on('end', () => {
-        client.end();
-        resolve(users);
-      });
-
-      query.on('error', (err) => reject(err));
-    });
+    return db.query(
+      sql,
+      [user.instrument, user.part, user.spotId, user.nameNumber, performanceId],
+      instanceFromRowChallengeableUser
+    );
   }
 
   static findAllForPerformanceCSV(performanceId) {
     const sql = queries.challengesForCSV;
 
-    return new Promise((resolve, reject) => {
-      const client = utils.db.createClient();
-      const challenges = [];
-
-      client.connect();
-      client.on('error', err => reject(err));
-
-      const query = client.query(sql, [performanceId]);
-
-      query.on('row', ({
-        challengee,
-        challenger,
-        challengeespot,
-        challengerspot,
-        firstnamenumber,
-        secondnamenumber,
-        spotopen
-      }) => {
-        challenges.push(new ChallengeForCSV(
-          challengee,
-          challenger,
-          challengeespot,
-          challengerspot,
-          firstnamenumber,
-          secondnamenumber,
-          spotopen
-        ));
-      });
-
-      query.on('end', () => {
-        client.end();
-        resolve(challenges);
-      });
-
-      query.on('error', err => {
-        client.end();
-        reject(err);
-      });
-    });
+    return db.query(sql, [performanceId], instanceFromRowChallengeForCSV);
   }
 
   static findAllRawForPerformance(performanceId) {
     const sql = 'SELECT * FROM challenges WHERE performanceId = $1';
```

Comment:
It just depended on how complicated the SQL was. Something simple like is fine as a string, but I preferred to write the SQL for bigger queries